index.bs

```diff
@@ -3303,58 +3358,74 @@ To <dfn>get the realm info</dfn> given |environment settings|:
+    <dt>|global object| is a {{SharedWorkerGlobalScope}} object
+    <dd>
+      1. Let |realm info| be a map matching the <code>SharedWorkerRealmInfo</code> production,
+        with the <code>realm</code> field set to |realm id|, the <code>type</code>
+        field set to "<code>shared-worker</code>" and the <code>origin</code> field set to |origin|.
 
-   <dd>
-      1. Let |type| be "<code>service-worker</code>".
+    <dt>|global object| is a {{ServiceWorkerGlobalScope}} object
+    <dd>
+      1. Let |realm info| be a map matching the <code>ServiceWorkerRealmInfo</code> production,
+        with the <code>realm</code> field set to |realm id|, the <code>type</code>
+        field set to "<code>service-worker</code>" and the <code>origin</code> field set to |origin|.
 
-   <dt>The [=Realm/global object=] specified by |environment settings| is a {{WorkerGlobalScope}} object
+   <dt>|global object| is a {{WorkerGlobalScope}} object
    <dd>
-      1. Let |type| be "<code>worker</code>".
+      1. Let |realm info| be a map matching the <code>WorkerRealmInfo</code> production,
+        with the <code>realm</code> field set to |realm id|, the <code>type</code>
+        field set to "<code>worker</code>" and the <code>origin</code> field set to |origin|.
 
-   <dt>The [=Realm/global object=] specified by |environment settings| is a {{PaintWorkletGlobalScope}} object
+   <dt>|global object| is a {{PaintWorkletGlobalScope}} object
    <dd>
-      1. Let |type| be "<code>paint-worklet</code>".
+      1. Let |realm info| be a map matching the <code>PaintWorkletRealmInfo</code> production,
+        with the <code>realm</code> field set to |realm id|, the <code>type</code>
+        field set to "<code>paint-worklet</code>" and the <code>origin</code> field set to |origin|.
 
-   <dt>The [=Realm/global object=] specified by |environment settings| is a {{AudioWorkletGlobalScope}} object
+   <dt>|global object| is a {{AudioWorkletGlobalScope}} object
    <dd>
-      1. Let |type| be "<code>audio-worklet</code>".
+      1. Let |realm info| be a map matching the <code>AudioWorkletRealmInfo</code> production,
+        with the <code>realm</code> field set to |realm id|, the <code>type</code>
+        field set to "<code>audio-worklet</code>" and the <code>origin</code> field set to |origin|.
 
-   <dt>The [=Realm/global object=] specified by |environment settings| is a {{WorkletGlobalScope}} object
+   <dt>|global object| is a {{WorkletGlobalScope}} object
    <dd>
-      1. Let |type| be "<code>worklet</code>".
+      1. Let |realm info| be a map matching the <code>WorkletRealmInfo</code> production,
+        with the <code>realm</code> field set to |realm id|, the <code>type</code>
+        field set to "<code>worklet</code>" and the <code>origin</code> field set to |origin|.
 
    <dt>Otherwise:
    <dd>
-      1. Return null.
+      1. Let |realm info| be a map matching the <code>GenericRealmInfo</code> production,
```

Comment:
I'm a bit worried about this. For example do we end up with an event for extension content script realms? It makes more sense to me to specifcally have a list of supported realm types.

Reply:
Removing GenericRealmInfo.